src/motion-queue.h

```diff
@@ -72,9 +75,27 @@ struct MotionSegment {
 #endif
 } __attribute__((packed));
 
+// Layout of the status register
+struct QueueStatus {
+  uint32_t counter : 24;
```

Comment:
These values need an explanation: what is the counter referring to and what is index referring to. Why is it sufficient that we can live with 24 bit here.
Right now, this might be fresh in your mind, but If they are not documented here, then someone else will have a hard time understanding the code ... or you yourself in 3 months after having worked on other parts.